From b109f2467c1a9f6853198126ac342e1bdf168597 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Wed, 14 Oct 2026 16:32:56 -0700
Subject: [PATCH] chronik: retain Dogecoin policy and genesis test fixtures

Port remaining narrow reference fixture deltas: Dogecoin chain-ID version bits, 88-DOGE genesis, 83-byte default data-carrier payloads or explicit 223-byte opt-in for token tests, and dogecoin.conf. Preserve all current tests and APIs.

Source: 9690833b8b6709b7a7837bc397f0d04bf3b885d1; original fixture lineage includes bd5b3624eb4389f887be4becfc9c6c673df1 and 3afd1a4d3a93e17faabc1f87414e6e4ddbfad87c.

diff --git a/test/functional/chronik_lokad_id_group.py b/test/functional/chronik_lokad_id_group.py
index 877bd2c78ebc2f977282c94bccef5d4cef04acb3..421d610d57d6f92385fc77ee9a15f663f6cb5b30 100644
--- a/test/functional/chronik_lokad_id_group.py
+++ b/test/functional/chronik_lokad_id_group.py
@@ -101,7 +101,7 @@ class ChronikLokadIdGroup(BitcoinTestFramework):
         node.setmocktime(mocktime + 1)
         tx1 = CTransaction()
         tx1.vin = [CTxIn(COutPoint(tx0.txid_int, 1), spend_p2lokad(b"lok1"))]
-        tx1.vout = [CTxOut(0, CScript([OP_RETURN, b"lok0", b"x" * 100]))]
+        tx1.vout = [CTxOut(0, CScript([OP_RETURN, b"lok0", b"x" * 70]))]
         chronik.broadcast_tx(tx1.serialize()).ok()
         assert_equal(lokad_id_unconf(b"lok0"), [tx0.txid_hex, tx1.txid_hex])
         assert_equal(lokad_id_unconf(b"lok1"), [tx1.txid_hex])
@@ -118,7 +118,7 @@ class ChronikLokadIdGroup(BitcoinTestFramework):
         tx2.vin = [CTxIn(COutPoint(tx0.txid_int, 2), spend_p2lokad(b"lok2"))]
         tx2.vout = [
             CTxOut(
-                0, CScript([OP_RETURN, OP_RESERVED, b"lok2__", b"lok0" + b"x" * 100])
+                0, CScript([OP_RETURN, OP_RESERVED, b"lok2__", b"lok0" + b"x" * 60])
             )
         ]
         chronik.broadcast_tx(tx2.serialize()).ok()
@@ -147,7 +147,7 @@ class ChronikLokadIdGroup(BitcoinTestFramework):
         tx3 = CTransaction()
         tx3.vin = [CTxIn(COutPoint(tx0.txid_int, 3), spend_p2lokad(b"lok3"))]
         tx3.vout = [
-            CTxOut(0, CScript([OP_RETURN, OP_RESERVED, b"lok2", b"lok0" + b"x" * 100]))
+            CTxOut(0, CScript([OP_RETURN, OP_RESERVED, b"lok2", b"lok0" + b"x" * 60]))
         ]
         chronik.broadcast_tx(tx3.serialize()).ok()
 
@@ -179,7 +179,7 @@ class ChronikLokadIdGroup(BitcoinTestFramework):
         tx3_conflict = CTransaction()
         tx3_conflict.vin = [CTxIn(COutPoint(tx0.txid_int, 3), spend_p2lokad(b"lok3"))]
         tx3_conflict.vout = [
-            CTxOut(0, CScript([OP_RETURN, OP_RESERVED, b"lok4" + b"x" * 100]))
+            CTxOut(0, CScript([OP_RETURN, OP_RESERVED, b"lok4" + b"x" * 60]))
         ]
 
         block = create_block(
diff --git a/test/functional/chronik_script_history.py b/test/functional/chronik_script_history.py
index b61b3e62c2ecc8c1fa0509bbb0b252d78dc7a9b2..beef20b7ad6bbac746f4eacf5cb8411264a7e8cd 100644
--- a/test/functional/chronik_script_history.py
+++ b/test/functional/chronik_script_history.py
@@ -10,7 +10,12 @@ from test_framework.address import (
     P2SH_OP_TRUE,
     SCRIPTSIG_OP_TRUE,
 )
-from test_framework.blocktools import GENESIS_CB_PK, create_block, create_coinbase
+from test_framework.blocktools import (
+    GENESIS_CB_PK,
+    VERSION_CHAIN_ID_BITS,
+    create_block,
+    create_coinbase,
+)
 from test_framework.messages import COutPoint, CTransaction, CTxIn, CTxOut
 from test_framework.p2p import P2PDataStore
 from test_framework.script import OP_RETURN, CScript
@@ -259,7 +264,7 @@ class ChronikScriptHistoryTest(BitcoinTestFramework):
             int(blockhashes[-1], 16),
             coinbase_tx,
             mocktime + 1100,
-            version=5,
+            version=VERSION_CHAIN_ID_BITS | 5,
             txlist=mine_txs,
         )
         block.solve()
diff --git a/test/functional/chronik_script_unconfirmed_txs.py b/test/functional/chronik_script_unconfirmed_txs.py
index 8f3120ae6a74d01e4bb9cf996dadb766e7f66905..6fa736dc525070b6645e4ea5d7ea92ccc2d71d92 100644
--- a/test/functional/chronik_script_unconfirmed_txs.py
+++ b/test/functional/chronik_script_unconfirmed_txs.py
@@ -118,7 +118,7 @@ class ChronikScriptUnconfirmedTxsTest(BitcoinTestFramework):
         for mocktime_offset in mocktime_offsets:
             cointxid = cointxids.pop(0)
             time_first_seen = mocktime + mocktime_offset
-            pad_script = CScript([OP_RETURN, bytes(100)])
+            pad_script = CScript([OP_RETURN, bytes(80)])
 
             tx = CTransaction()
             tx.nVersion = 1
diff --git a/test/functional/chronik_scripthash.py b/test/functional/chronik_scripthash.py
index 51861d3ffc133510b90aa112fe93c174695bbed5..cb4cf68220c859f39c01d1427845d97c4fb460bb 100644
--- a/test/functional/chronik_scripthash.py
+++ b/test/functional/chronik_scripthash.py
@@ -122,7 +122,7 @@ class ChronikScriptHashTest(BitcoinTestFramework):
                         ),
                         block_height=0,
                         is_coinbase=True,
-                        sats=50_000_000 * XEC,
+                        sats=88_000_000 * XEC,
                         is_final=False,
                     )
                 ],
diff --git a/test/functional/chronik_token_alp.py b/test/functional/chronik_token_alp.py
index 95d08959870e6956cbd91e7681bf6016831cf372..fa24a8f471288a49d39f85b31c56742b8b378c74 100644
--- a/test/functional/chronik_token_alp.py
+++ b/test/functional/chronik_token_alp.py
@@ -30,7 +30,7 @@ class ChronikTokenAlp(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-datacarriersize=223"]]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_chronik()
diff --git a/test/functional/chronik_token_burn.py b/test/functional/chronik_token_burn.py
index 54dd005a9dd7c3d9c6d096eb2a39f7540853fbb9..1c056aadf10dddf15e6d894222139cb4d2f70837 100644
--- a/test/functional/chronik_token_burn.py
+++ b/test/functional/chronik_token_burn.py
@@ -23,7 +23,7 @@ class ChronikTokenBurn(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-datacarriersize=223"]]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_chronik()
diff --git a/test/functional/chronik_token_id_group.py b/test/functional/chronik_token_id_group.py
index 91583932b68d67fb3f03f6f266364ef79eb8ddd1..d431876d6e1edd3a2b7c87c549343dfaaac89d2c 100644
--- a/test/functional/chronik_token_id_group.py
+++ b/test/functional/chronik_token_id_group.py
@@ -26,7 +26,7 @@ class ChronikTokenBurn(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-datacarriersize=223"]]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_chronik()
diff --git a/test/functional/chronik_token_slp_fungible.py b/test/functional/chronik_token_slp_fungible.py
index 38a543f90798705e5b2370f161c097a35f7a907d..1a265eab2503fd8388e729a2708261da291dec6e 100644
--- a/test/functional/chronik_token_slp_fungible.py
+++ b/test/functional/chronik_token_slp_fungible.py
@@ -22,7 +22,7 @@ class ChronikTokenSlpFungible(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-datacarriersize=223"]]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_chronik()
diff --git a/test/functional/chronik_token_slp_mint_vault.py b/test/functional/chronik_token_slp_mint_vault.py
index 9cc49acc7669c3f3bf6c341727a58c4c340e5ba8..ebdda8fd2a6eddb06da30522dd03327cf31e930b 100644
--- a/test/functional/chronik_token_slp_mint_vault.py
+++ b/test/functional/chronik_token_slp_mint_vault.py
@@ -26,7 +26,7 @@ class ChronikTokenSlpMintVault(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-datacarriersize=223"]]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_chronik()
diff --git a/test/functional/chronik_token_slp_nft1.py b/test/functional/chronik_token_slp_nft1.py
index ba48225c966f97b650a91a47625f04e731ef64a6..dde6369b448c1e2b5c06c8ae3baa9ad267ce2082 100644
--- a/test/functional/chronik_token_slp_nft1.py
+++ b/test/functional/chronik_token_slp_nft1.py
@@ -22,7 +22,7 @@ class ChronikTokenSlpNft1(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-datacarriersize=223"]]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_chronik()
diff --git a/test/functional/chronik_unconfirmed_txs.py b/test/functional/chronik_unconfirmed_txs.py
index 804bf1e96fbf5fd2169e007d21c5be86154dd123..41f1a874657519b09813f285008d999d6163eaf5 100644
--- a/test/functional/chronik_unconfirmed_txs.py
+++ b/test/functional/chronik_unconfirmed_txs.py
@@ -24,7 +24,7 @@ class ChronikUnconfirmedTxsTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-datacarriersize=223"]]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_chronik()
